### PR TITLE
chore(project): add SVGO script

### PR DESCRIPTION
**Pull request by LitoMore**
### Changes

We should install the `lint-staged` to the root.

- Move `lint-staged` package to root
- Add `lint:global` script for all files
- Update Node.js runtime to 16 for all workflows (See [this tweet](https://twitter.com/jeresuikkila/status/1440780658322726921))

_Prettier and ESLint can be moved to `lint:global` for linting JavaScript, TypeScript and JSON files. (Some files that are not in the workspace we haven't lint it yet.) We can do this in another PR._

### References

- [SVGO](https://github.com/okonet/lint-staged#how-to-use-lint-staged-in-a-multi-package-monorepo)
- [lint-staged](https://github.com/okonet/lint-staged)

### Hint

If you want to run svgo manually:

```shell
# Optimize all SVG files
yarn svgo -rf ./

# Optimize specific file
yarn svgo path/to/file.svg
```

### CODE REVIEW

**Comment by BlackHole1** [comment]
I don't recommend removing `lint-staged` from other packages. The reasons are as follows:
1. The dependencies of each package should be clearly visible, rather than having to look at its parent package.json
2. We are using the `yarn workspace` model, so declaring the same packages in packages does not increase hard disk space
3. This is a groundbreaking change as maintainers will not be able to find the relevant dependencies via the current package's `pacakge.json`. This makes the maintainer think that libraries installed elsewhere can be used in the current package.

For the reasons given in 1 3, I fear that in subsequent maintenance work, some developers will do the following
1. cd desktop/renderer-app/
2. yarn add A-Library
3. use `import {} from "A-Library"` in `web/flat-web` package

**While the above steps are fine in a local development environment, they will definitely fail in a CI environment**

This is because we remove packages that have not been modified in order to speed up the CI execution time and ensure that the `yarn install` is much faster when it is executed.

see: https://github.com/netless-io/flat/blob/3edb21b682cebc62c45cc60d60d544d92e6afa70/.github/workflows/deploy-web-dev.yml#L26-L27

**Comment by BlackHole1** [comment]
> _Prettier and ESLint can be moved to `lint:global` for linting JavaScript, TypeScript and JSON files. (Some files that are not in the workspace we haven't lint it yet.) We can do this in another PR._

On this point, it is probably also difficult to do, because the lint rules are not identical from package to package

see:

https://github.com/netless-io/flat/blob/3edb21b682cebc62c45cc60d60d544d92e6afa70/desktop/renderer-app/.lintstagedrc#L2-L5

https://github.com/netless-io/flat/blob/3edb21b682cebc62c45cc60d60d544d92e6afa70/packages/flat-i18n/.lintstagedrc#L2-L4

https://github.com/netless-io/flat/blob/3edb21b682cebc62c45cc60d60d544d92e6afa70/packages/flat-types/.lintstagedrc#L2-L5

**Comment by BlackHole1** [comment]
Of course, there's no denying that the other modifications to this PR are very good!

**Comment by LitoMore** [comment]
Like this PR https://github.com/netless-io/flat/pull/1090, I had to install the dependency for all workspace that really painful.

But I can add them back if you want. That's up to you.

**Comment by LitoMore** [comment]
> On this point, it is probably also difficult to do, because the lint rules are not identical from package to package

We can only move Prettier to the global.

**Comment by BlackHole1** [comment]
> Like this PR #1090, I had to install the dependency for all workspace that really painful.
> 
> But I can add them back if you want. That's up to you.

Although this process will be more difficult, one of the benefits it will bring is that each package will be easy to refactor/split in the future.

On how to improve the development experience at this point, I think we have two directions to start:

1. Push `yarn workspace` to support something like: `yarn workspace -p pacakges/A -p packages/B add A-Library`
2. Add Like `mech scripts`, see: https://firefox-source-docs.mozilla.org/mach/index.html
    图片

**Comment by BlackHole1** [comment]
> > On this point, it is probably also difficult to do, because the lint rules are not identical from package to package
> 
> We can only move Prettier to the global.

One of the goals of our project architecture is that any packages can be developed/run/build independently of anything else

_This is something we confirmed at the beginning of the project, as we had a lot of problems with development and refactoring before this rule existed._

File: svgo.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+    multipass: true,
+    eol: 'lf',
```
Comment (BlackHole1) [inline svgo.config.js]:
```suggestion
    eol: "lf",
```

File: package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
     "postinstall": "husky install",
     "bootstrap": "yarn run _init:agora-configure && yarn install --frozen-lockfile --force",
     "lint": "yarn workspaces run lint",
+    "lint:global": "lint-staged",
```
Comment (BlackHole1) [inline package.json]:
```suggestion
    "lint": "lint-staged && yarn workspaces run lint",
```
This would be better

**Review by BlackHole1 — CHANGES_REQUESTED** [review]

**Review by BlackHole1 — APPROVED** [review]
Great PR!

**Pull request merged**